### ISSUE TITLE
observation_period_end should default to the max transaction time

### ISSUE THREAD

**Issue by yanirs** [CamDavidsonPilon/lifetimes#67]
In `utils.calibration_and_holdout_data(transactions, customer_id_col, datetime_col, calibration_period_end, observation_period_end=datetime.today(), freq='D', datetime_format=None)`: `datetime.today()` is set once when the file is first read. This can be confusing when dealing with historical data, as the holdout column and evaluation results become dependent on the time the process is run.

It'd be better to default to `None` and then in the function body do:

``` python
if observation_period_end is None:
    observation_period_end = transactions[datetime_col].max()
```

**Comment by yanirs** [comment]
Actually, this applies to all functions that accept `observation_period_end` as an argument.

**Comment by CamDavidsonPilon** [comment]
Hm, I wonder about this. 

> holdout column and evaluation results become dependent on the time the process is run.

that's true, which isn't ideal, but I don't know if have the max observation as the period end is the best replacement. Interesting you brought this up.

**Comment by yanirs** [comment]
I hit this issue when I was changing `plot_calibration_purchases_vs_holdout_purchases` to calculate error metrics rather than just plot them (something that may be worth contributing back, but that's a separate issue). I use the X last days (`holdout_days`) of the dataset as a holdout set, so I thought it'd be enough to pass in the max date minus `holdout_days` as `calibration_period_end` and ignore the `observation_period_end` argument. However, when calling `plot_calibration_purchases_vs_holdout_purchases`, I found it unintuitive that `summary.iloc[0]['duration_holdout']` (on https://github.com/CamDavidsonPilon/lifetimes/blob/master/lifetimes/plotting.py#L59) is not equal to `holdout_days`. This affected the results when I replaced `summary.iloc[0]['duration_holdout']` with `holdout_days` in my code (which seemed like a sensible change, at least to me).

In general, I find it confusing that you get different results today and tomorrow if you run the example from the readme and omit the `observation_period_end` parameter:
```python
summary_cal_holdout = calibration_and_holdout_data(transaction_data, 'id', 'date', period_end='2014-09-01') 
bgf.fit(summary_cal_holdout['frequency_cal'], summary_cal_holdout['recency_cal'], summary_cal_holdout['T_cal'])
plot_calibration_purchases_vs_holdout_purchases(bgf, summary_cal_holdout)
```

In my opinion, the default value of `datetime.today()` is unlikely to be what users want (especially since it keeps changing). Do you have an idea for a better replacement than the max observation time?

**Comment by statwonk** [comment]
@CamDavidsonPilon, I have to add that I, too, expected the end of observation to depend only on my own data (that it'd use the `max(timestamp)`). I'm curious to hear what lead you to exposing that as an argument, I might have overlooked a use-case.

**Comment by CamDavidsonPilon** [comment]
This does seem to trip users up, and does make the analysis hard to replicate. I would be comfortable with changing the API to max(date)

**Comment by aprotopopov** [comment]
Fixed in #120
